3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -18,18 +19,67 @@ Complete the four functions provided in the starter `index.js` file:
 Try and avoid using global variables. Instead, use function parameters and 
 return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function fetchData(url) {
+  return fetch(url).then((response) => {
+    if (response.ok) {
+      return response.json();
+    }
+    throw new Error('Request failed');
+  });
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function fetchAndPopulatePokemons(data) {
+  const listOfPokemons = data.results;
+  const body = document.querySelector('body');
+  const buttonGetPokemon = document.createElement('button');
+  buttonGetPokemon.textContent = 'Get Pokemon!';
+  body.appendChild(buttonGetPokemon);
+  const select = document.createElement('select');
+  body.appendChild(select);
+  const showListOfPokemons = () => {
+    listOfPokemons.forEach((pokemon) => {
+      const option = document.createElement('option');
+      option.textContent = pokemon.name;
+      select.appendChild(option);
+    });
+    select.addEventListener('change', (event) =>
+      fetchImage(event.target, listOfPokemons)
+    );
+
+    buttonGetPokemon.removeEventListener('click', showListOfPokemons);
+  };
+  buttonGetPokemon.addEventListener('click', showListOfPokemons);
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function fetchImage(target, listOfPokemons) {
+  const currentPokemon = listOfPokemons.find((el) => el.name === target.value);
+
+  let imgSrc = '';
+  fetchData(currentPokemon.url)
+    .then((data) => {
+      imgSrc = data.sprites.front_default;
+      renderImg(imgSrc);
+    })
+    .catch((error) => console.log(error));
+  function renderImg(src) {
+    const body = document.querySelector('body');
+    let img = body.querySelector('img');
+    if (!img) {
+      img = document.createElement('img');
+      body.appendChild(img);
+    }
+    img.src = src;
+    img.alt = target.value;
+  }
 }
 
-function main() {
-  // TODO complete this function
+async function main() {
+  try {
+    const response = await fetchData('https://pokeapi.co/api/v2/pokemon/');
```

Comment:
As I mentioned, this fetch should take place in `fetchAndPopulatePokemons()` when the button is clicked.